src/controllers/suggestions.js

```diff
@@ -0,0 +1,385 @@
+    const fullResponse = {
+      suggestions: responses,
+      metadata: {
+        total: responses.length,
+        success: succeded,
+        failed: responses.length - succeded,
+      },
+    };
+
+    return createResponse(fullResponse, 207);
+  };
+
+  /**
+   * Updates data for a suggestion
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} the updated suggestion data
+   */
+  const patchSuggestion = async (context) => {
+    const siteId = context.params?.siteId;
+    const opportunityId = context.params?.opportunityId;
+    const suggestionId = context.params?.suggestionId;
+
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+
+    if (!hasText(opportunityId)) {
+      return badRequest('Opportunity ID required');
+    }
+
+    if (!hasText(suggestionId)) {
+      return badRequest('Suggestion ID required');
+    }
+
+    const suggestion = await Suggestion.findById(suggestionId);
+    if (!suggestion || suggestion.getOpportunityId() !== opportunityId) {
+      return notFound('Suggestion not found');
+    }
+    const opportunity = await suggestion.getOpportunity();
+    if (!opportunity || opportunity.getSiteId() !== siteId) {
+      return notFound('Suggestion not found');
+    }
+
+    // validate request body
+    if (!context.data) {
+      return badRequest('No updates provided');
+    }
+
+    let hasUpdates = false;
+    const { rank, data, kpiDeltas } = context.data;
+    try {
+      if (rank && rank !== suggestion.rank) {
+        hasUpdates = true;
+        suggestion.setRank(rank);
+      }
+
+      if (data) {
+        hasUpdates = true;
+        suggestion.setData(data);
+      }
+
+      if (kpiDeltas) {
+        hasUpdates = true;
+        suggestion.setKpiDeltas(kpiDeltas);
+      }
+
+      if (hasUpdates) {
+        const updatedSuggestion = await suggestion.save();
+        return ok(SuggestionDto.toJSON(updatedSuggestion));
+      }
+    } catch (e) {
+      return badRequest(e.message);
+    }
+    return badRequest('No updates provided');
+  };
+
+  /**
+   * Update the status of one or multiple suggestions in one transaction
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} the updated opportunity data
+   */
+  const patchSuggestionsStatus = async (context) => {
+    const siteId = context.params?.siteId;
+    const opportunityId = context.params?.opportunityId;
+
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+
+    if (!hasText(opportunityId)) {
+      return badRequest('Opportunity ID required');
+    }
+
+    // validate request body
+    if (!context.data) {
+      return badRequest('No updates provided');
+    }
+
+    if (!isArray(context.data)) {
+      return badRequest('Request body must be an array of [{ id: <suggestion id>, status: <suggestion status> },...]');
+    }
+
+    const suggestionPromises = context.data.map(async ({ id, status }, index) => {
```

Comment:
use `Suggestion.bulkUpdateStatus` instead?

Reply:
Agreed... will do in a separate PR